### PR TITLE
qemu/tests/live_snapshot_runtime.py: over-write action_when_start().

### PR DESCRIPTION
**Pull request by CongLi**
Do live snapshot during the pre-action, and need wait for pre-action complete when snapshot finished.

Signed-off-by: Cong Li <coli@redhat.com>

id: 1391925

### CODE REVIEW

**Comment by CongLi** [comment]
(1/1) smp_16.8192m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_scsi.up.virtio_net.Win7.i386.sp1.io-github-autotest-qemu.live_snapshot.with_payload.with_reboot: PASS (137.16 s)

**Comment by CongLi** [comment]
@vivianQizhu 

Could you please help review this PR?

Thanks.

**Comment by vivianQizhu** [comment]
With this modification, 'self.processes' is useless in whole Class, all related code can be removed.

**Comment by CongLi** [comment]
@vivianQizhu 
Thanks, code has been updated, could you help review it again?

Thanks.

**Comment by vivianQizhu** [comment]
Acked-by: Qianqian Zhu qizhu@redhat.com

File: qemu/tests/block_copy.py [hunk 1]
```diff
@@ -298,7 +297,10 @@ def action_when_start(self):
                 bg = utils.InterruptedThread(fun)
                 bg.start()
                 if bg.isAlive():
-                    self.processes.append(bg)
+                    try:
```
Comment (xutian) [inline qemu/tests/block_copy.py]: why try it

Comment (xutian) [inline qemu/tests/block_copy.py]: and join mean wait for the process finished that mean sub process not works in parallel not step by step. so I don't know why update it, case changed?

Comment (CongLi) [inline qemu/tests/block_copy.py]: For example:
1. system_reset
2. do live snapshot
3. verify alive

Current problem is that verify guest alive failed as reboot is not finished.

Comment (CongLi) [inline qemu/tests/block_copy.py]: For the try, just want to ignore all exceptions.

Comment (xutian) [inline qemu/tests/block_copy.py]: it looks not ignore, you raise it again, so that a useless try.

Comment (xutian) [inline qemu/tests/block_copy.py]: if your steps is 1, 2 then 3, your configuration should be action_when_start, 1, then 2, 3 in action_after/before_finished 3 in action_before_cleanup

**Comment by xutian** [comment]
if logical log orignal function mismatch your requests, right way it overwrite that function in your class

**Comment by CongLi** [comment]
Hi @xutian 

You mean the name of action_when_start() is not clear?
The func() is suitable for my request.

Thanks.

**Comment by CongLi** [comment]
813 17:27:28 INFO | PASS 1-smp_8.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.    virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.live_snapshot.with_payload.with_reboo    t

**Comment by CongLi** [comment]
@xutian 

Could you please help review it again?

Thanks.

**Comment by xutian** [comment]
LGTM.

**Pull request merged**